spec/git_helper_spec.rb

```diff
@@ -110,6 +110,74 @@
     end
   end
 
+  context('count_commits_between(base_ref:, head_ref:)') do
+    before do
+      # Spec branching setup:
+      #
+      #   (1.0)
+      # A---B---C         main
+      #        / \
+      #       |   D---E   feature-branch
+      #       F---G---H   another-branch / new-branch
```

Comment:
Lovely ASCII art 👍

Reply:
_(Though you're gonna have to redo it once you add the case for a merge commit aka commit with multiple parents to cover the case discussed in the other comment, sorry 😅 )_